src/ui/components/SecondaryToolbox/ReactDevTools.tsx

```diff
@@ -237,7 +238,7 @@ class ReplayWall implements Wall {
     }
   }
 
-  private async ensureReactDevtoolsBackendLoaded() {
+  public async ensureReactDevtoolsBackendLoaded() {
     if (this.recordingTarget === null) {
```

Comment:
This function will be called multiple times, and it _looks_ unsafe. (I think it's actually safe because our Suspense caches should prevent meaningful code from being double executed.) Still, I wonder if we should not _rely_ on that and add a better check here to only call it once.

For example, the way we're checking if `this.recordingTarget === null` doesn't prevent that code from executing multiple times (since we're awaiting an async value) and we don't even check before calling `injectReactDevtoolsBackend` again.

I know that neither of these should have any practical impact _now_ but it seems like the sort of thing that _could_ change over time.

Reply:
Not sure what you mean by "unsafe" in this case.  The worst that can happen here is that two calls to `ensureReactDevToolsBackendLoaded()` would run, the first kicks off the recording target request and waits, and the second one _tries_ to kick off the recording target request but it's already in flight.

Reply:
Really it's just saving us an additional `await` on later calls.

Reply:
By "unsafe" I meant that this method calls a function that looks like it could be (and maybe in the future, actually could be) side effectful– `injectReactDevtoolsBackend`. 

In general, I think it's good future-proofing pattern for "init"/"inject" type functions to be defensive against being called more than once.

Reply:
That was actually the point of using the eval cache - to ensure that it only actually runs once per pause.

(I could in theory do a sync check inside that to ask the cache if it has a cache entry for this pause at all, but I don't see how that would be much of an improvement over just waiting for the cache promise to resolve anyway.)

Reply:
Yeah, that's fair. I think the thing that concerns me is when I see code in one place relying on a particular implementation detail of code in another place that isn't really explicit in the name/api of that other place. But maybe I'm just over-worrying about this.

I agree with you that it shouldn't cause any problems in its current form. 👍🏼  Fair enough.